### PR TITLE
chore(executor): instrument errors to be more meaningful

### PR DESCRIPTION
**Pull request by rymnc**
## Linked Issues/PRs

Issues with `forc deploy` not too relevant without more meaningful errors. should help us debug.

## Description

- Updated the error type `ExecutorError::GasOverflow` with more details on what failed.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [ ] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

**Review by xgreenx — APPROVED** [review]

File: crates/types/src/services/executor.rs [hunk 1]
```diff
@@ -264,8 +264,10 @@ pub enum Error {
     OutputAlreadyExists,
     #[display(fmt = "The computed fee caused an integer overflow")]
     FeeOverflow,
-    #[display(fmt = "The computed gas caused an integer overflow")]
-    GasOverflow,
+    #[display(
+        fmt = "The computed gas caused an integer overflow. reason: {_0}, Augend: {_1}, Addend: {_2}"
```
Comment (AurelienFT) [inline crates/types/src/services/executor.rs:268]: Is it possible to inform about the limit (maybe for the user to retry within the limit)

Comment (rymnc) [reply crates/types/src/services/executor.rs:268]: the `Addend` should take care of that, in the case of `tx_max_gas > remaining_gas_limit`. maybe it is not too readable.

Comment (rymnc) [reply crates/types/src/services/executor.rs:268]: how does this work a532cc8b83 ?

**Review by AurelienFT — APPROVED** [review]
LGTM

File: crates/types/src/services/executor.rs (hunk 1 above)
Comment (AurelienFT) [reply crates/types/src/services/executor.rs:268]: Yeah find it more readable like this

**Review by acerone85 — APPROVED** [review]

**Pull request merged**